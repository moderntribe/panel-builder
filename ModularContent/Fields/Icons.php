<?php

namespace ModularContent\Fields;

/**
 * Class Icons
 *
 * @package ModularContent\Fields
 *
 * A field for selecting an icon from an icon library.
 */
class Icons extends Radio {

	protected $class_string       = '';
	protected $icon_prefix        = '';
	protected $font_size          = '';
	protected $label_size         = '';
	protected $search             = false;
	protected $show_uncategorized = false;
	protected $ajax_option        = false;
	protected $categories         = [];

	public function __construct( $args = [] ) {
		$this->defaults['strings']      = [
			'placeholder.search' => __( 'Search Icon Library', 'modular-content' ),
			'label.selected'     => __( 'Selected Icon:', 'modular-content' ),
		];
		$this->defaults['class_string']       = $this->class_string;
		$this->defaults['search']             = $this->search;
		$this->defaults['ajax_option']        = $this->ajax_option;
		$this->defaults['font_size']          = $this->font_size;
		$this->defaults['label_size']         = $this->label_size;
		$this->defaults['show_uncategorized'] = $this->show_uncategorized;
		$this->defaults['categories']         = $this->categories;
		$this->defaults['icon_prefix']        = $this->icon_prefix;
		parent::__construct( $args );
	}

	public function get_blueprint() {
		$blueprint                 = parent::get_blueprint();
		$options                   = $this->get_options();
		$blueprint['class_string'] = $this->class_string;
		$blueprint['search']       = $this->search;
		$blueprint['options']      = [];
		$blueprint['icon_prefix']  = $this->icon_prefix;
		$blueprint['font_size']    = $this->font_size;
		$blueprint['label_size']   = $this->label_size;

		foreach ( $options as $key => $label ) {

			$key = is_int( $key ) ? $label : $key; // convert non-associative array keys to the value instead.

			$blueprint['options'][] = [
				'label' => $label,
				'value' => (string) $key, // cast to string so react-select has consistent types for comparison
			];
		}
		$blueprint['ajax_option']         = $this->ajax_option;
		$blueprint['categories']          = $this->categories;
		$blueprint['show_uncategorized']  = $this->show_uncategorized;

		return $blueprint;
	}
}
