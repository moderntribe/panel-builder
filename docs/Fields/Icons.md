## Icons

A field for selecting an icon from an icon library.

### Parameters

||||
|---|---|---|
| `label`       | `string` | The __() translated label for this field       |
| `name`        | `string` | The unique name for this field                 |
| `description` | `string` | The __() translated description for this field |
| `options` | `array` | An array containing the unique classnames for each option |
| `search` | `bool` | Whether to display a search field for searching through the available options |
| `ajax_option` | `string` | The AJAX action name that will return the registered icon options |
| `font_size` | `string` | The font size (in px) to use for the icon previews |
| `label_size` | `string` | The label size (in px) to use for the icon label previews |
| `show_uncategorized` | `string` | Whether to show icons which are not assigned to a category |
| `categories` | `array` | An array of key => value pairs of Categories in which key is the category slug and value is the category Label |
| `icon_prefix` | `string` | An option prefix that should be added to each icon class string |
| `class_string` | `string` | The class string to be added to the element; %s will be replaced with the selected option |

### Example

```php
$field = new Icons( array(
    'label'              => __( 'List Icon' ),
    'name'               => 'list-icon',
    'description'        => __( 'The icon to use for each list item.' ),
    'options'            => [ 'fa-user', 'fa-access-denied', 'fa-list' ],
    'search'             => true,
    'class_string'       => 'fa %s',
    'ajax_option'        => 'get_fa_icons',
    'font_size'          => '12',
    'label_size'         => '10',
    'show_uncategorized' => true,
    'categories'         => [
        'medical' => __( 'Medical', 'tribe' ),
        'travel'  => __( 'Travel', 'tribe' ),
        'misc'    => __( 'Misc', 'tribe' ),
    ],
    'icon_prefix'        => 'fa-2x ',
) );
```