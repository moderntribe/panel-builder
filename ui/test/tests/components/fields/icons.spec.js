/* global expect b:true */

import React from 'react';
import { shallow, mount } from 'enzyme';
import Icons from 'components/fields/icons';
import renderer from 'react-test-renderer';

const props = {
	label: 'Arrow Icon',
	class_string: '%s',
	icon_prefix: 'fa-2x ',
	default: 'fas fa-arrow-alt-right',
	options: [
		'far fa-arrow-alt-circle-right',
		'fas fa-arrow-alt-circle-right',
		'fal fa-arrow-alt-circle-right',
		'far fa-arrow-alt-right',
		'fas fa-arrow-alt-right',
		'fal fa-arrow-alt-right',
		'far fa-arrow-alt-square-right',
		'fas fa-arrow-alt-square-right',
		'fal fa-arrow-alt-square-right',
		'far fa-arrow-alt-to-right',
		'fas fa-arrow-alt-to-right',
		'fal fa-arrow-alt-to-right',
		'far fa-arrow-circle-right',
		'fas fa-arrow-circle-right',
		'fal fa-arrow-circle-right',
		'far fa-arrow-right',
		'fas fa-arrow-right',
		'fal fa-arrow-right',
		'far fa-arrow-square-right',
		'fas fa-arrow-square-right',
		'fal fa-arrow-square-right',
		'far fa-arrow-to-right',
		'fas fa-arrow-to-right',
		'fal fa-arrow-to-right',
		'far fa-chevron-circle-right',
		'fas fa-chevron-circle-right',
		'fal fa-chevron-circle-right',
		'far fa-chevron-double-right',
		'fas fa-chevron-double-right',
		'fal fa-chevron-double-right',
		'far fa-chevron-right',
		'fas fa-chevron-right',
		'fal fa-chevron-right',
		'far fa-chevron-square-right',
		'fas fa-chevron-square-right',
		'fal fa-chevron-square-right',
		'far fa-long-arrow-alt-right',
		'fas fa-long-arrow-alt-right',
		'fal fa-long-arrow-alt-right',
		'far fa-long-arrow-right',
		'fas fa-long-arrow-right',
		'fal fa-long-arrow-right',
		'far fa-angle-double-right',
		'fas fa-angle-double-right',
		'fal fa-angle-double-right',
		'far fa-angle-right',
		'fas fa-angle-right',
		'fal fa-angle-right',
	],
	description: 'Select your arrow icon for the slider.',
};

describe('Icons field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Icons {...props} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<Icons />);
		expect(wrapper.type()).toEqual('div');
	});

	it('has a container for the icons.', () => {
		const wrapper = shallow(<Icons />);
		expect(wrapper.find('.panel-icon-container').length).toEqual(1);
	});

	it('renders a search field if that prop is true', () => {
		const wrapper = mount(<Icons search />);
		expect(wrapper.find('.panel-icon-search').length).toEqual(1);
	});
});
