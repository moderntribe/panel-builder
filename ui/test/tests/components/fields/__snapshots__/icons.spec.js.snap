exports[`Icons field renders correctly 1`] = `
<div
  className="field panel-field">
  <label
    className="label panel-field-label">
    Arrow Icon
    <span
      className="icon"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined} />
  </label>
  <div
    className="panel-icon-container container">
    0
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
    <label
      className="plicons-label iconLabel"
      style={Object {}}>
      <input
        checked={false}
        data-depth={0}
        data-field="icon-select"
        data-option-type="single"
        name="modular-content-"
        onChange={[Function]}
        type="radio"
        value="undefined" />
      <i
        className="fa-2x undefined icon"
        style={Object {}}
        title="undefined" />
    </label>
  </div>
</div>
`;
